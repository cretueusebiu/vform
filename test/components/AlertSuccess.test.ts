import { mount } from '@vue/test-utils'
import { Form, AlertSuccess } from 'vform'

describe('AlertSuccess', () => {
  test('render alert if successful', () => {
    const form = new Form({ username: '' })
    form.successful = true

    const wrapper = mount(AlertSuccess, {
      propsData: { form, message: 'Custom message' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render alert with slot', () => {
    const form = new Form({ username: '' })
    form.successful = true

    const wrapper = mount(AlertSuccess, {
      propsData: { form },
      slots: { default: '<div class="custom">Custom message</div>' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('deson\'t alert if has errors', () => {
    const form = new Form({ username: '' })
    form.errors.set({ username: 'Username is required' })

    const wrapper = mount(AlertSuccess, {
      propsData: { form, message: 'Custom message' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('dismiss alert', () => {
    const form = new Form({ username: '' })
    form.successful = true

    const wrapper = mount(AlertSuccess, {
      propsData: { form, message: 'Custom message' }
    })

    wrapper.find('.close').trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
