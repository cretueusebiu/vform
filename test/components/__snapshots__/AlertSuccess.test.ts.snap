// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertSuccess deson't alert if has errors 1`] = `""`;

exports[`AlertSuccess dismiss alert 1`] = `
<div role="alert" class="alert alert-success alert-dismissible"><button type="button" aria-label="Close" class="close"><span aria-hidden="true">×</span></button>
  <div>Custom message</div>
</div>
`;

exports[`AlertSuccess render alert if successful 1`] = `
<div role="alert" class="alert alert-success alert-dismissible"><button type="button" aria-label="Close" class="close"><span aria-hidden="true">×</span></button>
  <div>Custom message</div>
</div>
`;

exports[`AlertSuccess render alert with slot 1`] = `
<div role="alert" class="alert alert-success alert-dismissible"><button type="button" aria-label="Close" class="close"><span aria-hidden="true">×</span></button>
  <div class="custom">Custom message</div>
</div>
`;
